tools/chainbuilder/main.go

```diff
@@ -124,6 +126,23 @@ type BuilderConfig struct {
 }
 
 func Run(ctx context.Context, cfg BuilderConfig, dir string) error {
+	// Setup signal handling for graceful shutdown
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Goroutine to handle shutdown signals
+	go func() {
+		select {
+		case <-signalCh:
+			fmt.Println("\nReceived termination signal. Shutting down gracefully...")
```

Comment:
[question] why add a newline before this line?

```suggestion
			fmt.Printf("Received termination signal. Shutting down gracefully...\n")
```